_posts/2020-10-01-torch-network-from-scratch/torch_network_from_scratch.Rmd

```diff
@@ -0,0 +1,1076 @@
+---
+title: "Getting familiar with torch tensors"
+description: > 
+ In this first installment in a mini-series of four, we present the main things you will want to know about torch tensors. As an illustrative example, we'll code a simple neural network from scratch.
+
+author:
+  - name: Sigrid Keydana
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: keydanatorchnetworkfromscratch
+date: 10-01-2020
+categories:
+  - Torch
+  - R
+output:
+  distill::distill_article:
+    self_contained: false
+    toc: true
+preview: images/pic.jpg
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+Two days ago, I introduced [torch](https://github.com/mlverse/torch), an R package natively providing functionality that is brought to Python users by [PyTorch](https://pytorch.org/). In that post, I assumed basic familiarity with TensorFlow/Keras -- in all likelihood, not an unreasonable assumption in the context of this blog. Consequently, I portrayed `torch` in a way I figured would be helpful to someone who "grew up" with the Keras way of training a model: Aiming to focus on differences, yet not lose sight of the overall process.
+
+This post now changes perspective. We code a simple neural network "from scratch", making use of just a single of `torch`'s building blocks (a pretty major one, though): *tensors*. This network will be as "raw" (low-level) as can be. (For the less math-inclined people among us, it may serve as a refresher of what's actually going on beneath all those convenience tools they built for us. But the real purpose is to illustrate what can be done with tensors alone.)
+
+Subsequently, three posts (of increasingly shorter length) will show how to reduce the effort -- noticeably right from the start, enormously once we finish. At the end of this mini-series, you will have seen how automatic differentiation works in `torch`, how to use `module`s (layers, in `keras` speak, and compositions thereof), and optimizers. By then, you'll have a lot of the background desirable when applying `torch` to real-world tasks.
```

Comment:
```suggestion
Subsequently, three posts will progressively show how to reduce the effort -- noticeably right from the start, enormously once we finish. At the end of this mini-series, you will have seen how automatic differentiation works in `torch`, how to use `module`s (layers, in `keras` speak, and compositions thereof), and optimizers. By then, you'll have a lot of the background desirable when applying `torch` to real-world tasks.
```

Reply:
I think the length of the posts themselves isn't as relevant as the progressive ease (or helper-i-ness—I don't know a good word for it) of the content.

Reply:
yes you're right! you would never guess the real motivation of that qualification - it's as if i had to justify these getting shorter & shorter :-(